datafusion/common/Cargo.toml

```diff
@@ -59,6 +59,7 @@ object_store = { workspace = true, optional = true }
 parquet = { workspace = true, optional = true, default-features = true }
 pyo3 = { version = "0.20.0", optional = true }
 sqlparser = { workspace = true }
+tokio = { workspace = true }
```

Comment:
tokio dependency is added to the datafusion_common

Reply:
I also worry about this slightly, though the liklihood of someone using datafusion without tokio is likely low

I had two alternate ideas for consideration:
1.  Move to`datafusion_execution` crate (which would also then get a tokio dependence, but it is less widely used) 
2. Make a new crate to hold common execution pieces (e.g `datafusion_common_runtime` or `datafusion_common_tokio` or`datafusion_common_execution`) 

The new crate might seem like overkill initially, but I think there are several things that could probably go there to allow for better reuse. For example:
* https://github.com/apache/arrow-datafusion/blob/main/datafusion/physical-plan/src/stream.rs 
* https://github.com/apache/arrow-datafusion/blob/9aa01f35cf512e03196799c44cfecf7fe8af90e2/datafusion/physical-plan/src/common.rs#L99